setup.py

```diff
@@ -144,7 +166,7 @@ def check_for_openmp():
 EXTRA_COMPILE_ARGS = ['-O3', '-std=c++11', '-pedantic']
 EXTRA_LINK_ARGS = []
 
-if sys.platform == 'darwin':
+if sys.platform == 'darwin' and 'clang' in os.getenv('CC', 'cc'):
```

Comment:
removing this new check fixes the compilation problems. Not sure what a better way of detecting that we are using `clang` vs `gcc` is though as on OSX the default compiler is `clang` but it also responds to call to `gcc` etc. Only thing I can think of is looking at the output of `gcc --version`. Are there any people who use actual `gcc` on OSX?